examples/platform/silabs/main_task.h

```diff
@@ -0,0 +1,59 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *    Copyright (c) 2019 Google LLC.
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+/**********************************************************
+ * Includes
+ *********************************************************/
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include "cmsis_os2.h"
+#include "FreeRTOS.h"
+#include "event_groups.h"
+#include "task.h"
+int32_t wfx_wifi_rsi_init(void);
+
+#ifdef __cplusplus
+}
+#endif
+
+/**********************************************************
+ * Defines
+ *********************************************************/
+#define MAIN_TASK_STACK_SIZE (1024 * 8)

```

Comment:
can't all this be in `main.cpp `and we don't need the header. I know we discussed this before. But I still fail to see why the header is needed.

Reply:
Earlier I kept all this in main.cpp only 

But you told to create a header file , that's why i created it.

Reply:
This header file needs to be removed

Reply:
It got WiFi specific stuff (NOT generic) and it uses cmsis os while the rest of the platform stuff is FreeRTOS directly.

Reply:
cmsis is not a separate OS it is a wrapper for FreeRTOS API's only

third_party/silabs/gecko_sdk/util/third_party/freertos/cmsis/Source/cmsis_os2.c

Reply:
No Jean Cmsis os is not a separate OS , it is a wrapper around FreeRTSO API only
Check this file ( third_party/silabs/gecko_sdk/util/third_party/freertos/cmsis/Source/cmsis_os2.c)

And main_task.h is only included in main.cpp.

And it only includes the stuff related to task creation.

Reply:
> But you told to create a header file , that's why i created it. 
What I said was that you were duplicating those variables in multiple examples. (this was in slc when we didn't share 1 main for all apps. All apps were using their own main.cpp.) Now this isn't the case anymore. 

It's a different scenario now, since we have only 1 main, you will not be repeating the same variables init creation in all functions. Therefore it is cleaner to set it in the main directly and not in a separate header that is not used elsewhere.

You can keep the CMSIS OS call we will eventually transition all our apps to it. but let's get rid of the the header